src/libostree/ostree-sign-ed25519.c

```diff
@@ -157,6 +175,8 @@ ostree_sign_ed25519_data (OstreeSign *self, GBytes *data, GBytes **signature,
 
   size_t len;
   if (EVP_DigestSignInit (ctx, NULL, NULL, NULL, pkey)
+      && EVP_DigestSign (ctx, NULL, &len, g_bytes_get_data (data, NULL), g_bytes_get_size (data))
+      && (sig = g_malloc0 (len)) != NULL
```

Comment:
Allocating in the middle of a conditional feels weird. How about splitting the conditionals:

```
if (digestsigninit && digestsign)
  {
    g_autofree guchar *sig = g_malloc0 (len);
    if (digestsign)
      ...
  }
```

or so?